Lib/test/test_xmlrpc.py

```diff
@@ -1171,6 +1171,26 @@ def test_gzip_decode_limit(self):
         xmlrpclib.gzip_decode(encoded, max_decode=-1)
 
 
+class HeadersServerTestCase(BaseServerTestCase):
+    class RequestHandler(xmlrpc.server.SimpleXMLRPCRequestHandler):
+        test_header = None
+
+        def do_POST(self):
+            self.__class__.test_header = self.headers.get('x-test')
+            return super().do_POST()
+    requestHandler = RequestHandler
+
+    def setUp(self):
+        self.RequestHandler.test_header = None
+        return super().setUp()
+
+    def test_header(self):
```

Comment:
I can't say I really understand this test very well, but I do think I'd add other test cases, specifically the cases of setting more than one header and the case of an empty list.

Might also want to test what happens when you pass something other than a list, like a `tuple` or a `dict` for whatever behavior you expect. What about a `dict_items` object (e.g. `{'X-Test': 'foo'}.items()`)?